python/ray/tests/test_scheduling_2.py

```diff
@@ -304,6 +306,51 @@ def task2():
         assert set(ray.get(locations)) == worker_node_ids
 
 
+def test_demand_report_when_scale_up(shutdown_only):
```

Comment:
Does this test fail before the fix?

Reply:
Yes because the demand will always be 10.